### PR TITLE
Bump allowable torch version to be < 2.2

### PR DESCRIPTION
**Pull request by sashaDoubov**
# What does this PR do?
This commit raises the maximum permitted torch version to be < 2.2 for composer installation. 

# What issue(s) does this change relate to?

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def package_files(prefix: str, directory: str, extension: str):
     'torchmetrics>=0.10.0,<1.1',
     'torch_optimizer>=0.3.0,<0.4',
     'torchvision>=0.13.1,<0.17',
-    'torch>=1.13.1,<2.1.3',
+    'torch>=1.13.1,<2.2',
```
Comment (j316chuck) [inline setup.py:81]:
```suggestion
    'torch>=1.13.1,<2.2.1',
```

**Review by j316chuck — COMMENTED** [review]
I think it should be 2.2.1 but LGTm

**Comment by mvpatel2000** [comment]
@sashaDoubov close per discussion?

**Comment by sashaDoubov** [comment]
closing as per discussion with @mvpatel2000

**Pull request closed without merging**